libvast/src/detail/pid_file.cpp

```diff
@@ -0,0 +1,66 @@
+
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#include "vast/detail/pid_file.hpp"
+
+#include "vast/concept/parseable/numeric.hpp"
+#include "vast/concept/parseable/to.hpp"
+#include "vast/concept/printable/numeric.hpp"
+#include "vast/concept/printable/to_string.hpp"
+#include "vast/detail/assert.hpp"
+#include "vast/detail/posix.hpp"
+#include "vast/detail/system.hpp"
+#include "vast/error.hpp"
+
+#include <fcntl.h>
+#include <unistd.h>
+
+#include <sys/file.h>
+
+namespace vast::detail {
+
+caf::error acquire_pid_file(const path& filename) {
+  // If the file exists, we cannot continue.
+  if (exists(filename)) {
+    // Attempt to read file to display an actionable error message.
+    auto contents = load_contents(filename);
+    if (!contents)
+      return contents.error();
+    return caf::make_error(ec::filesystem_error,
+                           "stale PID file found: ", filename.str(),
+                           "terminate process", *contents,
+                           "or remove PID file manually");
```

Comment:
```suggestion
    return make_error(ec::filesystem_error,
                      "stale PID file found: ", filename.str(),
                      "terminate process", *contents,
                      "or remove PID file manually");
```
Formatting might be wrong with this one.

Reply:
Isn't the `make_error` overload going away with the next CAF release?